src/utils.tsx

```diff
@@ -443,7 +474,9 @@ export const generateUnitOutput = (
         .flat()
         .filter((sr) => sr.type === "ArmyBookRule"),
     ]),
-    "key"
+    (x) => {
+      return x.key || x.name;
+    }
```

Comment:
Basically, not all special rules have a `key` it turns out!

So by uniqing on key, we were actually ignoring the special rules that don't have a key.

Now we use name as a backup.